packages/desktop-client/src/browser-server.js

```diff
@@ -1,7 +1,36 @@
 /* globals importScripts, backend */
 let hasInitialized = false;
 
-self.addEventListener('message', e => {
+/**
+ * Sometimes the frontend build is way faster than backend.
+ * This results in the frontend starting up before backend is
+ * finished and thus the backend script is not available.
+ *
+ * The goal of this function is to retry X amount of times
+ * to retrieve the backend script with a small delay.
+ */
+const importScriptsWithRetry = async (script, { maxRetries = 5 } = {}) => {
+  try {
+    importScripts(script);
+  } catch (e) {
+    // Break if maxRetries has exceeded
+    if (maxRetries <= 0) {
+      throw e;
+    }
+
+    // Attempt to retry after a small delay
+    await new Promise(resolve =>
+      setTimeout(async () => {
+        await importScriptsWithRetry(script, {
+          maxRetries: maxRetries - 1,
+        });
+        resolve();
+      }, 5000),
```

Comment:
What is this based on? Having to wait up to 25 seconds to see an error message feels like a bad user experience. (And at least for me I would rather hit refresh than have to wait 5 whole seconds). Do you think it would make sense to use something like exponential backoff instead so that if the backend build finishes quickly it can be picked up before 5 seconds?

Reply:
I actually intentionally made this logic _stupid_. Didn't want to add too much complexity to this workaround. IMO ideally we should figure out a better way to bundle the API. This is only a temp. solution so people don't get confused when launching for the first time.

Reply:
That’s reasonable!